forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -176,6 +179,259 @@ class DSRG_MRPT2 : public MASTER_DSRG {
     /// Unitary matrix to block diagonal Fock
     ambit::BlockedTensor U_;
 
+    // NOTICE These are essential variables and functions for computing DSRG-MRPT2 gradient.
+    // Some variables may be redundant thus need further elimination
+    /// Set ambit tensor labels
+    void set_ambit_space();
+    /**
+     * Initialize tensors.
+     *
+     * Density tensors, one- and two-electron integrals, Fock matrix, DSRG and CI tensors.
+     */
+    void set_tensor();
+    /**
+     * Initializing the two-body density Gamma2_.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_density();
+    /**
+     * Initializing the Fock F.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_active_fock();
+    /**
+     * Initializing the one-electron integral H.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_h();
+    /**
+     * Initializing the ERIs V.
+     *
+     * NOTICE: this function shall be revoked in the future.
+     */
+    void set_v();
+    /// Set CI-relevant integrals
+    void set_ci_ints();
+    /**
+     * Initialize global variables.
+     *
+     * MO indices list, dimension of different MOs, CI-related variables etc.
+     */
+    void set_global_variables();
+    size_t nmo;
+    size_t ncore;
+    size_t nvirt;
+    size_t na;
+    size_t naux;
+    size_t nirrep;
+
+    const bool PT2_TERM         = true;
+    const bool X1_TERM          = true;
+    const bool X2_TERM          = true;
+    const bool X3_TERM          = true;
+    const bool X4_TERM          = true;
+    const bool X5_TERM          = true;
+    const bool X6_TERM          = true;
+    const bool X7_TERM          = true;
+    const bool CORRELATION_TERM = true;
+    /**
+     * Initializing the DSRG-related auxiliary tensors.
+     */
+    void set_dsrg_tensor();
+    /**
+     * Write the energy-weighted density matrix into Psi4 Lagrangian_.
+     */
+    void write_lagrangian();
+    /**
+     * Write spin_dependent one-RDMs coefficients into Psi4 Da_ and Db_.
+     *
+     * We assume "Da == Db". This function needs be changed if such constraint is revoked.
+     */
+    void write_1rdm_spin_dependent();
+    /**
+     * Write spin_dependent two-RDMs coefficients using IWL.
+     *
+     * Coefficients in d2aa and d2bb need be multiplied with additional 1/2!
+     * Specifically:
+     * If you have v_aa as coefficients before 2-RDMs_alpha_alpha, v_bb before
+     * 2-RDMs_beta_beta and v_bb before 2-RDMs_alpha_beta, you need to write
+     * 0.5 * v_aa, 0.5 * v_bb and v_ab into the IWL file instead of using
+     * the original coefficients v_aa, v_bb and v_ab.
+     */
+    void write_2rdm_spin_dependent();
+    /**
+     * Write density terms of (P|Q)^x and (pq|Q)^x, where P and Q are auxiliary basis.
```

Comment:
Do you mean something like this?
```suggestion
     * Write the density terms contracted with (P|Q)^x and (pq|Q)^x, where P and Q are auxiliary basis functions.
```